move-vm-backend/src/warehouse.rs

```diff
@@ -0,0 +1,82 @@
+use anyhow::Error;
+
+use hashbrown::HashMap;
+use move_core_types::account_address::AccountAddress;
+
+use move_core_types::identifier::Identifier;
+use move_core_types::language_storage::{ModuleId, StructTag};
+use move_core_types::resolver::{ModuleResolver, ResourceResolver};
+
+use crate::storage::Storage;
+
+/// Structure holding account data which is held under one Move address (or one key
+/// in Substrate storage).
+#[derive(Clone, Debug, Default)]
+pub struct AccountData {
+    /// Hashmap of the modules kept under this account.
+    pub modules: HashMap<Identifier, Vec<u8>>,
+    /// Hashmap of the resources kept under this account.
+    pub resources: HashMap<StructTag, Vec<u8>>,
+}
+
+/// Move VM storage implementation for Substrate storage.
+pub struct Warehouse<S: Storage> {
+    /// Substrate storage implementing Storage trait
+    storage: S,
+}
+
+impl<S: Storage> Storage for Warehouse<S> {
+    fn get(&self, key: &[u8]) -> Option<AccountData> {
+        self.storage.get(key)
+    }
```

Comment:
As we discussed, we won't need the Storage trait for Warehouse. We will remove it in the next PR

Reply:
Agree.